### PR TITLE
Fix double quote escaping in desktop entry (#4300)

### PR DESCRIPTION
**Pull request by l-pt**
In the Exec= value, double quotes (") inside a quoted argument must be escaped with two backslashes (just like the dollar sign).
Moreover, this fixes the following errors, also reported in the issue, when launching electrum.desktop with kioclient5 (KDE plasma program runner):
```
$ kioclient exec /usr/share/applications/electrum.desktop 
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 6: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
kf.config.core: "KConfigIni: In file /usr/share/applications/electrum.desktop, line 21: " "Invalid escape sequence \"\\\"\"."
```

I have tested this modification with kioclient5 version 5.27.4 and gtk-launch version 3.24.37, with both launchers electrum is starting correctly without errors and with $HOME/.local/bin prepended to $PATH.

Relevant parts of the freedesktop Desktop Entry Specification:

[https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html)
>  Arguments may be quoted in whole. If an argument contains a reserved character the argument must be quoted. The rules for quoting of arguments is also applicable to the executable name or path of the executable program as provided.

> Quoting must be done by enclosing the argument between double quotes and escaping the double quote character, backtick character ("\`"), dollar sign ("$") and backslash character ("\\") by preceding it with an additional backslash character. Implementations must undo quoting before expanding field codes and before passing the argument to the executable program. Reserved characters are space (" "), tab, newline, double quote, single quote ("'"), backslash character ("\\"), greater-than sign (">"), less-than sign ("<"), tilde ("~"), vertical bar ("|"), ampersand ("&"), semicolon (";"), dollar sign ("$"), asterisk ("*"), question mark ("?"), hash mark ("#"), parenthesis ("(") and (")") and backtick character ("\`").

> Note that the general escape rule for values of type string states that the backslash character can be escaped as ("\\\\") as well and that this escape rule is applied before the quoting rule. As such, to unambiguously represent a literal backslash character in a quoted argument in a desktop entry file requires the use of four successive backslash characters ("\\\\\\\\"). Likewise, a literal dollar sign in a quoted argument in a desktop entry file is unambiguously represented with ("\\\\$").

### CODE REVIEW

**Comment by l-pt** [comment]
AppImage build failing due to a desktop-file-validate bug that was fixed 2 years ago:
[https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/commit/63840764746e5d7be502c50ee87c5f0b0c42d24b](https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/commit/63840764746e5d7be502c50ee87c5f0b0c42d24b)

**Comment by SomberNight** [comment]
related:
https://github.com/spesmilo/electrum/pull/4166 and https://github.com/spesmilo/electrum/issues/4300
https://github.com/spesmilo/electrum/issues/8004

**Comment by l-pt** [comment]
AppImage build is now passing

It seems the problem was that AppImageTool bundles the desktop-file-validate program, and the bundled version  (0.24) in AppImageTool 13 is buggy (see [here](https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/commit/63840764746e5d7be502c50ee87c5f0b0c42d24b)).

The bundled version [has been updated](https://github.com/AppImage/AppImageKit/pull/1204) but there have not been any more releases of AppImageTool other than the [continuous build](https://github.com/AppImage/AppImageKit/releases/tag/continuous) which we cannot verify the sha256sum of.

Since we already have to extract AppImageKit to patch mksquashfs it seems reasonable to just substitute the bundled version of desktop-file-validate with the newer version 0.26 that does not have known validation bugs. Let me know.

**Comment by SomberNight** [comment]
Thanks for looking into this.
As an alternative, what do you think about reverting to pre-https://github.com/spesmilo/electrum/pull/4166, like https://github.com/spesmilo/electrum/issues/8004 suggests? i.e. removing the PATH-manipulation

File: contrib/build-linux/appimage/make_appimage.sh [hunk 1]
```diff
@@ -40,10 +41,26 @@ verify_hash "$CACHEDIR/functions.sh" "8f67711a28635b07ce539a9b083b8c12d5488c0000
...
+tar xf "$CACHEDIR/desktop-file-utils-$DESKTOP_FILE_UTILS_VERSION.tar.xz" -C "$CACHEDIR"
+(
+    if [ -f "$CACHEDIR/desktop-file-utils-$DESKTOP_FILE_UTILS_VERSION/src/desktop-file-validate" ]; then
+        info "desktop-file-utils already built, skipping"
+        exit 0
+    fi
+    cd "$CACHEDIR/desktop-file-utils-$DESKTOP_FILE_UTILS_VERSION"
+
+    ./autogen.sh --silent
+    make "-j$CPU_CPOUNT" -s
```
Comment (SomberNight) [inline contrib/build-linux/appimage/make_appimage.sh:62]: typo `$CPU_CPOUNT`

**Comment by l-pt** [comment]
> Thanks for looking into this. As an alternative, what do you think about reverting to pre-#4166, like #8004 suggests? i.e. removing the PATH-manipulation

Seems right, after a brief Github search I couldn't find any other python projects changing the PATH in the desktop file

**Comment by SomberNight** [comment]
Ok. Good to hear; as I am much more in favour of removing it than keep hacking on top.
Would you like to create another PR that removes the PATH manipulation? Or should I do it?
As the content of this PR is non-trivial, I'd rather keep it just in case and not have it force-pushed.

**Pull request closed without merging**